test/OpenTelemetry.Instrumentation.Hangfire.Tests/HangfireInstrumentationJobFilterAttributeTests.cs

```diff
@@ -149,6 +150,46 @@ public async Task Should_Create_Activity_With_Custom_DisplayName()
         Assert.Equal(ActivityKind.Internal, activity.Kind);
     }
 
+    [Theory]
+    [InlineData("null", true)]
+    [InlineData("true", true)]
+    [InlineData("false", false)]
+    [InlineData("throw", false)]
+    public async Task Should_Respect_Filter_Option(string filter, bool shouldRecord)
+    {
+        // Arrange
+        Action<HangfireInstrumentationOptions> configure = filter switch
+        {
+            "null" => options => options.Filter = null,
+            "true" => options => options.Filter = _ => true,
+            "false" => options => options.Filter = _ => false,
+            "throw" => options => options.Filter = _ => throw new Exception("Filter throws exception"),
+            _ => throw new ArgumentOutOfRangeException(nameof(filter), filter, "Unexpected value"),
+        };
+
+        var processedItems = new List<Activity>();
+        var activityProcessor = new Mock<BaseProcessor<Activity>>();
+        activityProcessor
+            .Setup(p => p.OnStart(It.IsAny<Activity>()))
+            .Callback<Activity>(processedItems.Add);
+
+        using var tel = Sdk.CreateTracerProviderBuilder()
+            .AddHangfireInstrumentation(configure)
+            .AddProcessor(activityProcessor.Object)
+            .Build();
+
+        // Act
+        var jobId = BackgroundJob.Enqueue<TestJob>(x => x.Execute());
+        await this.WaitJobProcessedAsync(jobId, 5);
+
+        // Assert
+        Assert.Single(processedItems);
+        var activity = processedItems.First();
+
+        Assert.Equal(shouldRecord, activity.IsAllDataRequested);
+        Assert.Equal(shouldRecord, activity.ActivityTraceFlags.HasFlag(ActivityTraceFlags.Recorded));
```

Comment:
Side note: I found two tests in EF Core that
```c#
Assert.True(activity.ActivityTraceFlags.HasFlag(ActivityTraceFlags.None));
```

Which is wrong because it will always return `true`. It should be:
```c#
Assert.False(activity.ActivityTraceFlags.HasFlag(ActivityTraceFlags.Recorded));
```

Reply:
Could you please create PR with the fix?